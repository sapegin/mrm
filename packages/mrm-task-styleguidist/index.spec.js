jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const fs = jest.requireActual('fs');
const path = require('path');
const { install } = require('mrm-core');
const { omitBy } = require('lodash');
const vol = require('memfs').vol;
const task = require('./index');

const stringify = o => JSON.stringify(o, null, '  ');

afterEach(() => {
	vol.reset();
	install.mockClear();
});

it('should add React Styleguidist', () => {
	vol.fromJSON({
		[`${__dirname}/templates/styleguide.config.js`]: fs
			.readFileSync(path.join(__dirname, 'templates/styleguide.config.js'))
			.toString(),
		'/package.json': stringify({
			name: 'unicorn',
		}),
	});

	task({});

	expect(
		omitBy(vol.toJSON(), (v, k) => k.startsWith(__dirname))
	).toMatchSnapshot();
	expect(install.mock.calls).toEqual([
		[['react', 'react-dom'], { dev: false }],
		[['react-styleguidist', 'webpack']],
	]);

	// Check again with already set `start` script
	task(getConfigGetter({}));

	expect(
		omitBy(vol.toJSON(), (v, k) => k.startsWith(__dirname))
	).toMatchSnapshot();
	expect(install.mock.calls).toEqual([
		[['react', 'react-dom'], { dev: false }],
		[['react-styleguidist', 'webpack']],
		[['react', 'react-dom'], { dev: false }],
		[['react-styleguidist', 'webpack']],
	]);
});

it('should use a TypeScript template for a TypeScript project', () => {
	vol.fromJSON({
		[`${__dirname}/templates/styleguide.config.typescript.js`]: fs
			.readFileSync(
				path.join(__dirname, 'templates/styleguide.config.typescript.js')
			)
			.toString(),
		'/package.json': stringify({
			name: 'unicorn',
			devDependencies: {
				typescript: '*',
			},
		}),
	});

	task({});

	expect(vol.toJSON()['/styleguide.config.js']).toMatchSnapshot();
});

it('should not install webpack when used with CRA', () => {
	vol.fromJSON({
		[`${__dirname}/templates/styleguide.config.js`]: fs
			.readFileSync(path.join(__dirname, 'templates/styleguide.config.js'))
			.toString(),
		'/package.json': stringify({
			name: 'unicorn',
			dependencies: {
				'react-scripts': '*',
			},
		}),
	});

	task({});

	expect(install.mock.calls).toEqual([
		[['react', 'react-dom'], { dev: false }],
		[['react-styleguidist']],
	]);
});
