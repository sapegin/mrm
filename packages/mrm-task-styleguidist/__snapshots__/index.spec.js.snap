// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add React Styleguidist 1`] = `
Object {
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"styleguide\\": \\"styleguidist server\\",
    \\"styleguide:build\\": \\"styleguidist build\\",
    \\"start\\": \\"styleguidist server\\"
  }
}",
  "/styleguide.config.js": "module.exports = {
	components: 'src/components/**/[A-Z]*.js',
};
",
}
`;

exports[`should add React Styleguidist 2`] = `
Object {
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"styleguide\\": \\"styleguidist server\\",
    \\"styleguide:build\\": \\"styleguidist build\\",
    \\"start\\": \\"styleguidist server\\"
  }
}",
  "/styleguide.config.js": "module.exports = {
	components: 'src/components/**/[A-Z]*.js',
};
",
}
`;

exports[`should use a TypeScript template for a TypeScript project 1`] = `
"module.exports = {
	components: 'src/components/**/[A-Z]*.tsx',
	propsParser: require('react-docgen-typescript').parse,
};
"
`;
